bottles/backend/repos/repo.py

```diff
@@ -88,24 +86,17 @@ def get_manifest(self, url: str, plain: bool = False):
             return {}
 
 class RepoStatus:
-    LOCKS: Dict[str, PyLock] = {}
-
-    @staticmethod
-    def repo_start_operation(name: str):
-        lock = RepoStatus.LOCKS.setdefault(name, PyLock())
-        lock.acquire()
-        logging.debug(f"Start operation {name}")
+    EVENTS: Dict[str, PyEvent] = {}
 
     @staticmethod
     def repo_done_operation(name: str):
-        lock = RepoStatus.LOCKS.setdefault(name, PyLock())
-        if lock.locked():
-            lock.release()
+        event = RepoStatus.EVENTS.setdefault(name, PyEvent())
+        event.set()
         logging.debug(f"Done operation {name}")
 
+    @staticmethod
     def repo_wait_operation(name: str):
-        lock = RepoStatus.LOCKS.setdefault(name, PyLock())
-        logging.debug(f"Wait operation {name}")
-        lock.acquire()
-        lock.release()
-        logging.debug(f"Done wait operation {name}")
\ No newline at end of file
+        event = RepoStatus.EVENTS.setdefault(name, PyEvent())
+        # By default when an Event is created, it will be false, so it will block
+        event.wait()
```

Comment:
I don't think they are idempotent
it will wait for event forever if the operation you wait ended too quick or never start
the execution order of async functions is hard to ensure

Reply:
I see what you mean. One solution could be to check if the operation existed and finished, and if yes, simply continuing? That would mean that this operation could only be "waited" once, let's say. It would work for the use case that we have, but then it's a problem when/if we generalize this class.

Reply:
> One solution could be to check if the operation existed and finished

that's exactly what I mentioned before in Discord by "busy wait is the only option for this if you don’t want to introduce any other variable to record the fetching started state"

anyway this solution works for me :D

Reply:
My bad, the `wait` method won't wait forever, just ignore this comment
```
wait(timeout=None) will block until the internal flag is true. If the internal flag is true on entry, return immediately.
```